apis/services.d.ts

```diff
@@ -290,6 +291,10 @@ interface OnEventHandler {
   (req: Request, next: Function): Promise<any> | any | void
 }
 
+interface OnErrorHandler {
+  (err: Error, req: Request): any | void
```

Comment:
Perfectly in line with all our other handler signatures, so this is absolutely fine. I am just wondering: can we stricten the return type for handlers in any way in general? Can it maybe be derived from the parameters?

Reply:
The return type does not matter for before/after/on('error').
Promises are awaited.

For regular on-handlers the return type might be undefined or an instance or array of "Books" or a promise resolving to one of them.

I don't know if that's somehow expressable.

Reply:
Yes, we should be able to express this using [`TypedRequest<T>`](https://github.com/cap-js/cds-types/blob/main/apis/services.d.ts#L315). It would then look more or less like this:

```ts
interface FooHandler<T> {
  (..., req: TypedRequest<T>): T | Promise<T> 
}
```
which would basically mean "T in, T out". But lets postpone this into a separate issue, as it seems to be beyond the scope of your actual fix.